lib/silo/commands/file_list.rb

```diff
@@ -55,10 +55,43 @@ def run
         data = silo.list(dir)
         dirs, files = data['directories'], data['files']
 
-        dirs&.each do |dir|
-          puts Paint[bold(dir), :blue]
+        dirs.map! { |d| { 'type' => 'd', 'name' => d } }
+        files.map! { |f| { 'type' => 'f', 'name' => f[:name] } }
+        list = dirs.concat(files).sort_by { |i| i['name'].downcase }
+        number_of_items = list.size
+        screen_width = IO.console.winsize[1]
+        number_of_rows = 1
+
+        print_width = nil
+        column_items = nil
+        loop do
+          column_items = list.each_slice(number_of_rows).to_a
+          print_width = 0
+          column_items.map! do |cis|
+            column_width = cis.max_by { |ci| ci['name'].length }['name'].length + 2
+            print_width += column_width
+            {
+              'width' => column_width,
+              'items' => cis
+            }
+          end
+          break if print_width <= screen_width
+          number_of_rows += 1
+        end
+
+        number_of_rows.times do |i|
+          output_row = ""
+          column_items.each do |cis|
+            column_width = cis['width']
+            if cis['items'][i]
+              type = cis['items'][i]['type']
+              name = cis['items'][i]['name']
+              name += " " * (column_width - name.length)
+              output_row += type == 'd' ? Paint[bold(name), :blue] : name
+            end
```

Comment:
It's slightly better code style to use `next unless cis['items'][i]` here. Fewer indentations makes the code easier to read by other developers.